webapp/src/components/SimulatorPresenceBar.tsx

```diff
@@ -19,7 +19,14 @@ function PlayerPresenceIcon(props: React.PropsWithoutRef<{slot: 1 | 2 | 3 | 4}>)
             type: "setactiveplayer",
             playerNumber: slot,
         };
+        const connectionMsg: pxsim.multiplayer.ConnectionMessage = {
+            type: "multiplayer",
+            content: "Connection",
+            slot: slot,
+            connected: true,
+        };
         simulator.driver.postMessage(setSlotMsg);
+        simulator.driver.postMessage(connectionMsg);
```

Comment:
If the player is already connected then I don't think we should raise the event again. Connects and disconnects should be reciprocal (like React's mount/unmount, etc.)

Reply:
That's handled in the controller connection -- the game engine listens for this message and sets https://github.com/microsoft/pxt-common-packages/blob/master/libs/game/controller.ts#L256, which raises the controller event only when it has actually changed

Reply:
Oh gotcha! Thanks.